src/DictVectors/dvec.jl

```diff
@@ -1,113 +1,156 @@
-    DVec(key => value; capacity)
-    DVec(args...; capacity)
-    DVec(d::Dict [, capacity])
-    DVec(v::Vector{V} [, capacity])
-Dictionary-based vector-like data structure with minimum capacity `capacity`
-for storing values with keys.
-The type of the values is `eltype(dv) == V`.
-Indexing is done with an
-arbitrary (in general non-integer) key with `keytype(dv) == K`.
-If the keyword argument `capacity` is passed then args are parsed as for `Dict`.
-When constructed from a `Vector`,
-the keys will be integers ∈ `[1, length(v)]`. See [`AbstractDVec`](@ref). The
-method [`capacity()`](@ref) is defined but not a strict upper limit as `Dict`
-objects can expand.
-"""
-struct DVec{A,T} <: AbstractDVec{A,T}
-    d::Dict{A,T}
-end
-# default constructor from `Dict` just wraps the dict, no copying or allocation:
-# dv = DVec(Dict(k => v, ...))
+    DVec{K,V,D<:AbstractDict{K,V},S}
 
-# constructor like Dict with mandatory keyword capacity
-DVec(args...; capacity) = DVec(Dict(args...), capacity)
+Dictionary-based vector-like data structure for use with FCIQMC and
+[KrylovKit](https://github.com/Jutho/KrylovKit.jl). While mostly behaving like a `Dict`, it
+supports various linear algebra operations such as `norm` and `dot`. It has a
+[`StochasticStyle`](@ref) that is used to select an appropriate spawning strategy in the
+FCIQMC algorithm.
 
-function DVec(dict::D, capacity::Int) where D <: Dict
-    sizehint!(dict, capacity)
-    return DVec(dict)
-end
+See also: [`AbstractDVec`](@ref).
+
+# Constructors
+
+* `DVec(dict::AbstractDict[, style, capacity])`: create a `DVec` with `dict` for storage.
+  Note that the data may or may not be copied.
+
+* `DVec(args...[; style, capacity])`: `args...` are passed to the `Dict` constructor. The
+  `Dict` is used for storage.
+
+* `DVec{K,V}([style, capacity])`: create an empty `DVec{K,V}`.
+
+* `DVec(dv::AbstractDVec[, style, capacity])`: create a `DVec` with the same contents as
+   `adv`. The `style` is inherited from `dv` by default.
 
-# by specifying keytype, eltype, and capacity
-DVec{K,V}(capacity::Int) where V where K = DVec(Dict{K,V}(), capacity)
+The default `style` is selected based on the `DVec`'s `valtype` (see
+[`default_style`](@ref)). If a style is given and the `valtype` does not match the `style`'s
+`eltype`, the values are converted to an appropriate type.
 
-# from Vector
-function DVec(v::AbstractVector{T}, capacity = length(v)) where T
-    indices = 1:length(v) # will be keys of dictionary
-    ps = map(tuple,indices,v) # create iterator over pairs
-    return DVec(Dict(ps), capacity)
+The capacity argument is optional and sets the initial size of the `DVec` via `sizehint!`.
+
+# Examples
+
+```jldoctest
+julia> dv = DVec(:a => 1)
+DVec{Symbol,Int64,IsStochasticInteger} with 1 entries
+  :a => 1
+
+julia> dv = DVec(:a => 2, :b => 3; style=IsDynamicSemistochastic())
+DVec{Symbol,Float32,IsDynamicSemistochastic{true}} with 2 entries
+  :a => 2.0f0
+  :b => 3.0f0
```

Comment:
What you get with current code is different:
```suggestion
julia> dv = DVec(:a => 2, :b => 3; style=IsDynamicSemistochastic())
DVec{Symbol,Float64} with 2 entries, style = IsDynamicSemistochastic{true}(1.0, Inf, 1.0)
  :a => 2.0
  :b => 3.0
```